opm/simulators/utils/readDeck.cpp

```diff
@@ -235,33 +233,31 @@ namespace {
                       const std::optional<int>&            outputInterval,
                       Opm::ErrorGuard&                     errorGuard)
     {
-        if (((deck == nullptr) || (schedule == nullptr) || (summaryConfig == nullptr)) &&
+        if (((schedule == nullptr) || (summaryConfig == nullptr)) &&
             (parseContext == nullptr))
         {
             OPM_THROW(std::logic_error,
-                      "We need a parse context if deck, schedule, "
+                      "We need a parse context if schedule "
                       "or summaryConfig are not initialized");
         }
 
         auto parser = Opm::Parser{};
-        if (deck == nullptr) {
-            deck = readDeckFile(deckFilename, checkDeck, parser,
-                                *parseContext, errorGuard);
-        }
+        auto deck = readDeckFile(deckFilename, checkDeck, parser,
+                                 *parseContext, errorGuard);
```

Comment:
Can we mark this as `const`?  If so, I think we should.

Reply:
indeed.